src/ogc/gml/GmlRectifiedGrid.js

```diff
@@ -0,0 +1,218 @@
+                    this.axisNames = this.axisNames || [];
+                    this.axisNames.push(child.textContent);
+                } else if (child.localName === "origin") {
+                    this.origin = this.assembleOrigin(child);
+                } else if (child.localName === "offsetVector") {
+                    this.offsetVector = this.offsetVector || [];
+                    this.offsetVector.push(this.assembleOffsetVector(child));
+                }
+            }
+        };
+
+        GmlRectifiedGrid.prototype.captureAttributes = function (element) {
+            this.id = element.getAttribute("gml:id");
+            this.srsName = element.getAttribute("srsName");
+            this.srsDimension = element.getAttribute("srsDimension");
+            this.axisLabels = element.getAttribute("axisLabels");
+            if (this.axisLabels) {
+                this.axisLabels = this.axisLabels.split(/\s+/);
+            }
+            this.uomLabels = element.getAttribute("uomLabels");
+            if (this.uomLabels) {
+                this.uomLabels = this.uomLabels.split(/\s+/);
+            }
+            this.dimension = parseInt(element.getAttribute("dimension"));
+        };
+
+        GmlRectifiedGrid.prototype.assembleLimits = function (element) {
+            var children = element.children || element.childNodes;
+
+            for (var c = 0; c < children.length; c++) {
+                var child = children[c];
+
+                if (child.localName === "GridEnvelope") {
+                    return this.assembleGridEnvelope(child);
+                }
+            }
+        };
+
+        GmlRectifiedGrid.prototype.assembleGridEnvelope = function (element) {
+            var children = element.children || element.childNodes, envelop = {};
+
+            for (var c = 0; c < children.length; c++) {
+                var child = children[c];
+                envelop[child.localName] = child.textContent.split(/\s+/);
+            }
+
+            return envelop;
+        };
+
+        GmlRectifiedGrid.prototype.assembleOrigin = function (element) {
+            var origin = {};
+            origin.type = element.getAttribute("xlink:type");
+            origin.href = element.getAttribute("xlink:href");
+            origin.role = element.getAttribute("xlink:role");
+            origin.arcrole = element.getAttribute("xlink:arcrole");
+            origin.title = element.getAttribute("xlink:title");
+            origin.show = element.getAttribute("xlink:show");
+            origin.actuate = element.getAttribute("xlink:actuate");
+            origin.nilReason = element.getAttribute("nilReason");
+            origin.remoteSchema = element.getAttribute("gml:remoteSchema");
+            origin.owns = element.getAttribute("owns");
+
+            var children = element.children || element.childNodes;
+
+            for (var c = 0; c < children.length; c++) {
+                var child = children[c];
```

Comment:
Dave pointed out the other day that the form:
for (var c = 0, len=children.length; c < len; c++)
is preferable over:
for (var c = 0; c < children.length; c++)
for performance reasons.

Reply:
I agree, and usually use that approach with for loops; however, the legacy (WMS, WMTS) XML parsing code uses the pattern applied in WCS. I opted for consistency over performance.